app/assets/javascripts/Components/Result/file_viewer.jsx

```diff
@@ -112,57 +112,34 @@ export class FileViewer extends React.Component {
     });
 
     this.setState({loading: true, url: null}, () => {
-      if (!this.props.selectedFileURL) {
-        fetch(
-          Routes.get_file_course_submission_path(this.props.course_id, this.props.submission_id, {
-            submission_file_id: submission_file_id,
-            force_text: force_text,
-          }),
-          {credentials: "include"}
-        )
-          .then(res => res.json())
-          .then(body => {
-            if (body.type === "image" || body.type === "pdf" || this.isNotebook(body.type)) {
-              this.setState({type: body.type}, () => {
-                this.setFileUrl(submission_file_id);
-              });
-            } else {
-              const content = JSON.parse(body.content).replace(/\r?\n/gm, "\n");
-              this.setState({
-                content: content,
-                type: body.type,
-                loading: false,
-              });
-            }
-          });
+      this.setFileUrl(submission_file_id);
+
+      if (
+        this.props.selectedFileType === "image" ||
+        this.props.selectedFileType === "pdf" ||
+        this.isNotebook(this.props.selectedFileType)
+      ) {
+        this.setState({type: this.props.selectedFileType}, () => {
+          this.setFileUrl(submission_file_id);
+        });
       } else {
-        if (
-          this.props.selectedFileType === "image" ||
-          this.props.selectedFileType === "pdf" ||
-          this.isNotebook(this.props.selectedFileType)
-        ) {
-          this.setState({type: this.props.selectedFileType}, () => {
-            this.setFileUrl();
-          });
-        } else {
-          const requestData = {preview: true, force_text: force_text};
-          const url = this.props.selectedFileURL;
-          const queryString = new URLSearchParams(requestData);
-          const requestUrl = `${url}&${queryString}`;
-          fetch(requestUrl)
-            .then(response => {
-              if (response.ok) {
-                return response.text();
-              }
+        const delimiter = this.props.selectedFileURL.includes("?") ? "&" : "?";
```

Comment:
This delimiter-handling part is quite awkward; I believe you're running into this because you're trying to unify the handling of multiple different URLs.

Rather than insert this here, please modify *how the URLs are originally generated* to include these query parameters directly. (I assume this will involve a few different locations, because of the different places this component is used.)